backend/src/main/java/org/eclipse/tractusx/puris/backend/delivery/logic/adapter/DeliveryInformationSammMapper.java

```diff
@@ -0,0 +1,168 @@
+
+import org.eclipse.tractusx.puris.backend.common.domain.model.measurement.ItemQuantityEntity;
+import org.eclipse.tractusx.puris.backend.delivery.domain.model.EventTypeEnumeration;
+import org.eclipse.tractusx.puris.backend.delivery.domain.model.OwnDelivery;
+import org.eclipse.tractusx.puris.backend.delivery.domain.model.ReportedDelivery;
+import org.eclipse.tractusx.puris.backend.delivery.logic.dto.deliverysamm.Delivery;
+import org.eclipse.tractusx.puris.backend.delivery.logic.dto.deliverysamm.DeliveryInformation;
+import org.eclipse.tractusx.puris.backend.delivery.logic.dto.deliverysamm.Location;
+import org.eclipse.tractusx.puris.backend.delivery.logic.dto.deliverysamm.OrderPositionReference;
+import org.eclipse.tractusx.puris.backend.delivery.logic.dto.deliverysamm.Position;
+import org.eclipse.tractusx.puris.backend.delivery.logic.dto.deliverysamm.TransitEvent;
+import org.eclipse.tractusx.puris.backend.delivery.logic.dto.deliverysamm.TransitLocations;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Material;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Partner;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialPartnerRelationService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+public class DeliveryInformationSammMapper {
+    @Autowired
+    private MaterialPartnerRelationService mprService;
+
+    public DeliveryInformation ownDeliveryToSamm(List<OwnDelivery> deliveryList) {
+        if (deliveryList == null || deliveryList.isEmpty()) {
+            log.warn("Can't map empty list");
+            return null;
+        }
+        Partner partner = deliveryList.get(0).getPartner();
+        if (deliveryList.stream().anyMatch(deli -> !deli.getPartner().equals(partner))) {
+            log.warn("Can't map delivery list with different partners");
+            return null;
+        }
+        Material material = deliveryList.get(0).getMaterial();
+        if (deliveryList.stream().anyMatch(deli -> !deli.getMaterial().equals(material))) {
+            log.warn("Can't map delivery list with different materials");
+            return null;
+        }
+        var groupedByPositionAttributes = deliveryList
+                .stream()
+                .collect(Collectors.groupingBy(prod -> new PositionsMappingHelper(
+                    prod.getCustomerOrderNumber(),
+                    prod.getSupplierOrderNumber(),
+                    prod.getCustomerOrderPositionNumber()
+                )));
+        DeliveryInformation samm = new DeliveryInformation();
+
+        samm.setMaterialGlobalAssetId(material.getMaterialNumberCx());
```

Comment:
I think we need a slightly new concept here because there are two dimensions:
- partner (customer & supplier)
- incoterm steering more or less who offers the data

Thus we can't just always use the material's cx number (I'm acting as supplier). From my understanding it would be sufficient to enhance signature to also take in the + a direction. For stock this was not needed as we had the differentiation into material and product